src/cell.cpp

```diff
@@ -110,14 +110,100 @@ tokenize(const std::string region_spec) {
   return tokens;
 }
 
+//==============================================================================
+//! Add precedence for infix regions so intersections have higher
+//! precedence than unions using parentheses.
+//==============================================================================
+
+vector<int32_t>::iterator add_parentheses(
+  vector<int32_t>::iterator start, vector<int32_t>& infix)
+{
+  int32_t start_token = *start;
+  // Add left parenthesis
+  if (start_token == OP_INTERSECTION) {
+    start = infix.insert(start - 1, OP_LEFT_PAREN);
+  } else {
+    start = infix.insert(start + 1, OP_LEFT_PAREN);
+  }
+  start++;
+
+  // Initialize return iterator
+  auto return_iterator = infix.begin();
```

Comment:
There were some further changes in here from openmc-dev/openmc#2318 that should be included.

Reply:
I've added in the changes. There's a few very minor differences owing to the OpenMP port not having the `Region` abstraction yet, but is generally 1:1.